discovery/db_discovery.go

```diff
@@ -19,38 +18,52 @@ import (
 	"github.com/golang/glog"
 )
 
-var cacheRefreshInterval = 1 * time.Hour
-var getTicker = func() *time.Ticker {
-	return time.NewTicker(cacheRefreshInterval)
+type ticketParamsValidator interface {
+	ValidateTicketParams(ticketParams *pm.TicketParams) error
+}
+
+type roundsManager interface {
+	LastInitializedRound() *big.Int
 }
 
 type DBOrchestratorPoolCache struct {
-	node *core.LivepeerNode
+	store                 common.OrchestratorStore
+	lpEth                 eth.LivepeerEthClient
+	ticketParamsValidator ticketParamsValidator
+	rm                    roundsManager
 }
 
-func NewDBOrchestratorPoolCache(node *core.LivepeerNode) *DBOrchestratorPoolCache {
-	if node.Eth == nil {
-		glog.Error("Could not refresh DB list of orchestrators: LivepeerNode nil")
+func NewDBOrchestratorPoolCache(store common.OrchestratorStore, lpEth eth.LivepeerEthClient, rm roundsManager, ticketParamsValidator ticketParamsValidator) *DBOrchestratorPoolCache {
+	if store == nil {
+		glog.Error("could not create DBOrchestratorPoolCache: store is nil")
 		return nil
 	}
 
-	_ = cacheRegisteredTranscoders(node)
+	if lpEth == nil {
+		glog.Error("could not create DBOrchestratorPoolCache: LivepeerNode is nil")
```

Comment:
Error message should be updated to not say "LivepeerNode"

Reply:
fixed in 74f273ef